openapi-converter/clients_schema_to_openapi/src/main.rs

```diff
@@ -0,0 +1,176 @@
+//     tracing::subscriber::set_global_default(subscriber)?;
+//
+//     clients_schema_to_openapi::convert_schema_file(
+//         "../output/schema/schema-no-generics.json",
+//         None,
+//         |e| true,
+//         //|e| e.name == "search",
+//         //|e| e.name == "clear_scroll",
+//         //|e| e.name.starts_with("indices."),
+//         //|e| e.name == "indices.exists_alias" || e.name == "indices.delete_alias",
+//         //|e| &e.name[0..1] < "d",
+//         std::fs::File::create("../output/openapi/elasticsearch-openapi.json")?
+//         //std::io::stdout()
+//     )?;
+//
+//     Ok(())
+// }
+
+impl Cli {
+    fn run(self) -> anyhow::Result<()>{
+
+        let json = if self.schema == Path::new("-") {
+            std::io::read_to_string(std::io::stdin())?
+        } else {
+            std::fs::read_to_string(self.schema)?
+        };
+
+        let mut model: clients_schema::IndexedModel = serde_json::from_str(&json)?;
+
+        if let Some(flavor) = self.flavor {
+            if flavor != SchemaFlavor::All {
+                let filter: fn(&Option<Availabilities>) -> bool = match flavor {
+                    SchemaFlavor::All => |_| {
+                        true
+                    },
+                    SchemaFlavor::Stack => |a| {
+                        // Generate public and private items for Stack
+                        clients_schema::Flavor::Stack.available(a)
+                    },
+                    SchemaFlavor::Serverless => |a| {
+                        // Generate only public items for Serverless
+                        clients_schema::Flavor::Serverless.visibility(a) == Some(Visibility::Public)
+                    },
+                };
+
+                model = clients_schema::transform::expand_generics(model)?;
+                model = clients_schema::transform::filter_availability(model, filter)?;
+            }
+        }
+
+        let openapi = clients_schema_to_openapi::convert_schema(&model)?;
+
+        let output: Box<dyn std::io::Write> = {
+            if let Some(output) = self.output {
+                if output == Path::new("-") {
+                    Box::new(std::io::stdout())
+                } else {
+                    Box::new(std::fs::File::create(output)?)
+                }
+            } else {
+                Box::new(std::io::stdout())
+            }
+        };
+        let output = std::io::BufWriter::new(output);
+
+        serde_json::to_writer_pretty(output, &openapi)?;
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser)]
+#[command(author, version, about, long_about = None)]
+pub struct Cli {
+    /// Schema file to transform. Use '-' for stdin.
+    schema: PathBuf,
+
+    #[arg(short, long)]
+    /// Output file. Defaults to stdout.
+    output: Option<PathBuf>,
+
+    /// Elasticsearch flavor to produce
+    #[arg(short, long)]
+    flavor: Option<SchemaFlavor>
+}
+
+#[derive(Debug, Clone, PartialEq, ValueEnum)]
+pub enum SchemaFlavor {
```

Comment:
Because there's we're likely to have a `Common` code generation target in the future (currently commented-out) that will be used to retain what's in both Stack and Serverless. So `client_schema::Flavor` represents the `@availability` annotation whereas here this represents the generation flavor that can be different from a simple XOR between Stack and Serverless.